arch/x86/x86.c

```diff
@@ -28,7 +28,6 @@ Z_INTERNAL int x86_cpu_has_sse42;
 Z_INTERNAL int x86_cpu_has_pclmulqdq;
 Z_INTERNAL int x86_cpu_has_vpclmulqdq;
 Z_INTERNAL int x86_cpu_has_tzcnt;
-Z_INTERNAL int x86_cpu_well_suited_avx512;
```

Comment:
I would prefer that the removal of cpu_well_suited_avx512 stuff is in a separate commit.